vortex-scalar/src/datafusion.rs

```diff
@@ -111,9 +110,11 @@ impl TryFrom<Scalar> for ScalarValue {
                             }
                         },
                     });
+                } else {
+                    // Unknown extension type: perform scalar conversion using the canonical
+                    // scalar DType.
+                    ScalarValue::try_from(Scalar::new(ext.scalars_dtype().clone(), value))?
```

Comment:
Same here, we need to delegate it to extension dtype

Reply:
ah, ok we should have set of extension dtypes that map to all arrow dtypes then your storage dtype is one of those extension dtypes